### PR TITLE
[Cellular Library] unit test for cellular_at_core.c

### PR DESCRIPTION
**Pull request by andysun2015**
*Description of changes:*
- Add CMock submodule for unit test
- Add unit test case for cellular_at_core.c

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tools/cmock/create_test.cmake [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+# Run the C preprocessor on target files.
+# Takes a CMAKE list of arguments to pass to the C compiler
+function(preprocess_mock_list mock_name file_list compiler_args)
+    set_property(GLOBAL PROPERTY ${mock_name}_processed TRUE)
+    foreach (target_file IN LISTS file_list)
+        # Has to be TARGET ALL so the file is pre-processed before CMOCK
+        # is executed on the file.
+        add_custom_command(OUTPUT ${target_file}.backup
+            COMMAND scp ${target_file} ${target_file}.backup
+            VERBATIM COMMAND ${CMAKE_C_COMPILER} -E ${compiler_args} ${target_file} > ${target_file}.out
```
Comment (paulbartell) [inline tools/cmock/create_test.cmake:67]: Typically preprocessed files have the a .i extension.
You could probably use -o rather than > to handle the output.

File: common/src/cellular_at_core.c [hunk 2]
```diff
@@ -68,11 +68,6 @@ static CellularATError_t validateString( const char * pString,
     CellularATError_t atStatus = CELLULAR_AT_SUCCESS;
     size_t stringLength = 0U;
 
-    if( ( pString == NULL ) || ( pStringValidationResult == NULL ) )
```
Comment (paulbartell) [inline common/src/cellular_at_core.c:71]: With this change, validateString will only ever return CELLULAR_AT_SUCCESS. Is it necessary to pStringValidationResult via a pointer or could the validateString function just return pStringValidationResult?

File: common/src/cellular_at_core.c [hunk 3]
```diff
@@ -111,21 +105,21 @@ CellularATError_t Cellular_ATIsPrefixPresent( const char * pString,
     char * ptrPrefixChar = NULL;
     char * ptrChar = NULL;
 
-    if( pResult == NULL )
+    if( pResult == NULL || pString == NULL )
```
Comment (paulbartell) [inline common/src/cellular_at_core.c:108]: Similar question with this function. Are separate pResult and atStatus parameters / return values really necessary?

File: test/unit-test/cellular_at_core_utest.c [hunk 4]
```diff
@@ -0,0 +1,932 @@
...
+    cellularStatus = Cellular_ATRemovePrefix( NULL );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+}
+
+/**
+ * @brief Test happy path for Cellular_ATRemovePrefix to return CELLULAR_AT_SUCCESS and got expected results.
+ */
+void test_Cellular_ATRemovePrefix_Happy_Path( void )
+{
+    CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
```
Comment (paulbartell) [inline test/unit-test/cellular_at_core_utest.c:206]: This intermediate variable is not necessary.

File: test/unit-test/cellular_at_core_utest.c [hunk 5]
```diff
@@ -0,0 +1,932 @@
...
+    char * pString = CELLULAR_SAMPLE_PREFIX_STRING_INPUT;
```
Comment (paulbartell) [inline test/unit-test/cellular_at_core_utest.c:207]: It's very hard to follow this test case when CELLULAR_SAMPLE_PREFIX_STRING_INPUT and CELLULAR_SAMPLE_PREFIX_STRING_OUTPUT are defined as macros. Defining the string inside the test case would greatly ease readability.

File: test/unit-test/cellular_at_core_utest.c [hunk 6]
```diff
@@ -0,0 +1,932 @@
...
+
+    /* Check that CELLULAR_BAD_PARAMETER is returned if the parameter ppString set to sample string is passed. */
```
Comment (paulbartell) [inline test/unit-test/cellular_at_core_utest.c:209]: Can you clarify this comment?

File: common/src/cellular_at_core.c [hunk 7]
```diff
@@ -300,7 +310,15 @@ CellularATError_t Cellular_ATRemoveTrailingWhiteSpaces( char * pString )
     char * p = NULL;
     CellularATStringValidationResult_t stringValidationResult = CELLULAR_AT_STRING_UNKNOWN;
 
-    atStatus = validateString( pString, &stringValidationResult );
+    if ( pString == NULL )
```
Comment (paulbartell) [inline common/src/cellular_at_core.c:313]: What is the scope of these parsing functions? Should they be private rather than public?

File: tools/cmock/create_test.cmake (hunk 1 above)
Comment (andysun2015) [reply tools/cmock/create_test.cmake:67]: I referred to this from [coreMQTT ](https://github.com/FreeRTOS/coreMQTT/blob/f756d9833ad79b60084def59935fdabf475c08b0/tools/cmock/create_test.cmake#L67).
Does this not correct?

File: common/src/cellular_at_core.c (hunk 2 above)
Comment (andysun2015) [reply common/src/cellular_at_core.c:71]: Yes, this change makes the CELLULAR_AT_SUCCESS meaningless.
Let me fix this changes.

File: common/src/cellular_at_core.c (hunk 3 above)
Comment (andysun2015) [reply common/src/cellular_at_core.c:108]: Yes, this change makes the CELLULAR_AT_SUCCESS meaningless.
Let me fix this changes.

File: test/unit-test/cellular_at_core_utest.c (hunk 6 above)
Comment (andysun2015) [reply test/unit-test/cellular_at_core_utest.c:209]: sure.

File: test/unit-test/cellular_at_core_utest.c (hunk 5 above)
Comment (andysun2015) [reply test/unit-test/cellular_at_core_utest.c:207]: Referring to [coreMQTT unit-test](https://github.com/FreeRTOS/coreMQTT/blob/main/test/unit-test/core_mqtt_utest.c), I think the macro should be much more readable?

File: tools/cmock/create_test.cmake (hunk 1 above)
Comment (paulbartell) [reply tools/cmock/create_test.cmake:67]: It is correct either way.

File: common/src/cellular_at_core.c [hunk 8]
```diff
@@ -56,49 +56,38 @@ typedef enum CellularATStringValidationResult
...
     size_t stringLength = 0U;
 
-    if( ( pString == NULL ) || ( pStringValidationResult == NULL ) )
+    /* The strnlen() function returns strlen(s), if that is less than maxlen,
+        * or maxlen if there is no null terminating ('\0') among the first
+        * maxlen characters pointed to by s.
+        *
+        * stringLength == CELLULAR_AT_MAX_STRING_SIZE is valid because it means that
+        * ( CELLULAR_AT_MAX_STRING_SIZE + 1 ) character is null terminating
+        * character.*/
```
Comment (aggarg) [inline common/src/cellular_at_core.c:76]:
```suggestion
    /* The strnlen() function returns strlen(s), if that is less than maxlen,
     * or maxlen if there is no null terminating ('\0') among the first
     * maxlen characters pointed to by s.
     *
     * stringLength == CELLULAR_AT_MAX_STRING_SIZE is valid because it means that
     * ( CELLULAR_AT_MAX_STRING_SIZE + 1 ) character is null terminating
     * character.*/
```

File: common/src/cellular_at_core.c [hunk 9]
```diff
@@ -760,11 +777,11 @@ CellularATError_t Cellular_ATStrtoi( const char * pStr,
     }
     else
     {
-        /* MISRA C 2012 Directive 4.7 � If a function returns error information, then
+        /* MISRA C 2012 Directive 4.7 � If a function returns error information, then
```
Comment (aggarg) [inline common/src/cellular_at_core.c:780]: is this intentional?

File: test/unit-test/CMakeLists.txt [hunk 10]
```diff
@@ -0,0 +1,60 @@
+# Include filepaths for source and include.
+# include( ${MODULE_ROOT_DIR}/mqttFilePaths.cmake )
```
Comment (aggarg) [inline test/unit-test/CMakeLists.txt:2]: Please remove the commented out code.

File: test/unit-test/cellular_at_core_utest.c [hunk 11]
```diff
@@ -0,0 +1,932 @@
...
+
+#include "cellular_at_core.h"
+
+/**
+ * @brief Cellular sample prefix string input.
+ */
+#define CELLULAR_SAMPLE_PREFIX_STRING_INPUT               "+CPIN:READY"
+
+/**
+ * @brief Cellular sample prefix string wron input.
```
Comment (aggarg) [inline test/unit-test/cellular_at_core_utest.c:44]:
```suggestion
 * @brief Cellular sample prefix string wrong input.

```

File: test/unit-test/cellular_at_core_utest.c [hunk 12]
```diff
@@ -0,0 +1,932 @@
...
+
+/**
+ * @brief Cellular sample string trailing white space.
+ */
+#define CELLULAR_SAMPLE_STRING_TRAILING_WHITE_SPACE        "response    "
+
+/**
+ * @brief Cellular sample string all white space.
+ */
+#define CELLULAR_SAMPLE_STRING_ALL_WHITE_SPACE             "    response    "
```
Comment (aggarg) [inline test/unit-test/cellular_at_core_utest.c:71]: `CELLULAR_SAMPLE_STRING_ALL_WHITE_SPACE --> CELLULAR_SAMPLE_STRING_LEADING_AND_TRAILING_WHITE_SPACE`?

File: test/unit-test/cellular_at_core_utest.c [hunk 13]
```diff
@@ -0,0 +1,932 @@
...
+ */
+#define CELLULAR_SAMPLE_TOKEN_OUTPUT                       "TEST_TOKEN:TOKEN1"
+
+/**
+ * @brief Cellular sample string delimter first.
+ */
+#define CELLULAR_SAMPLE_STRING_DELIMTER_FIRST              ",TEST_TOKENTOKEN1TOKEN2"
+
+/**
+ * @brief Cellular sample string delimter first.
```
Comment (aggarg) [inline test/unit-test/cellular_at_core_utest.c:104]:
```suggestion
 * @brief Cellular sample string wrong delimiter first.

```

File: test/unit-test/cellular_at_core_utest.c [hunk 14]
```diff
@@ -0,0 +1,932 @@
...
+ */
+#define CELLULAR_SAMPLE_TOKEN_STRING_INPUT                 "TEST_TOKEN:TOKEN1,TOKEN2,TOKEN3"
+
+/**
+ * @brief Cellular sample token string output.
+ */
+#define CELLULAR_SAMPLE_TOKEN_OUTPUT                       "TEST_TOKEN:TOKEN1"
+
+/**
+ * @brief Cellular sample string delimter first.
```
Comment (aggarg) [inline test/unit-test/cellular_at_core_utest.c:99]:
```suggestion
 * @brief Cellular sample string delimiter first.

```

File: test/unit-test/cellular_at_core_utest.c [hunk 15]
```diff
@@ -0,0 +1,932 @@
...
+ */
+#define CELLULAR_SAMPLE_STRTOL_HAPPY_CASE_STRING           "20210487"
+
+/**
+ * @brief Cellular sample strtol happy case number.
+ */
+#define CELLULAR_SAMPLE_STRTOL_HAPPY_CASE_NUMBER           ( 20210487 )
+
+/**
+ * @brief Cellular sample prefix string wron input.
```
Comment (aggarg) [inline test/unit-test/cellular_at_core_utest.c:154]:
```suggestion
 * @brief Cellular sample prefix string wrong input.

```

File: test/unit-test/cellular_at_core_utest.c [hunk 16]
```diff
@@ -0,0 +1,932 @@
...
+    TEST_ASSERT_EQUAL( CELLULAR_AT_SUCCESS, cellularStatus );
+
+    /* Test the result to check its functionality. */
+    TEST_ASSERT_EQUAL_STRING( CELLULAR_SAMPLE_PREFIX_STRING_OUTPUT, pString );
+}
+
+/**
+ * @brief Test the error path for Cellular_ATRemovePrefix to return CELLULAR_AT_BAD_PARAMETER.
+ */
+void test_Cellular_ATRemovePrefix_Error_Path( void )
```
Comment (aggarg) [inline test/unit-test/cellular_at_core_utest.c:220]: I'd break it in 3 tests:
* test_Cellular_ATRemovePrefix_Empty_String
* test_Cellular_ATRemovePrefix_Too_Large_String
* test_Cellular_ATRemovePrefix_Invalid_String

File: test/unit-test/cellular_at_core_utest.c [hunk 17]
```diff
@@ -0,0 +1,932 @@
...
+    TEST_ASSERT_EQUAL( CELLULAR_AT_SUCCESS, cellularStatus );
+
+    /* Test the result to check its functionality. */
+    TEST_ASSERT_EQUAL_STRING( CELLULAR_SAMPLE_STRING_NO_WHITE_SPACE, pString );
+}
+
+/**
+ * @brief Test the error path for Cellular_ATRemoveLeadingWhiteSpaces to return CELLULAR_AT_BAD_PARAMETER.
+ */
+void test_Cellular_ATRemoveLeadingWhiteSpaces_Error_Path( void )
```
Comment (aggarg) [inline test/unit-test/cellular_at_core_utest.c:271]: Error_Path seems to generic. How about Empty_String instead?

File: test/unit-test/cellular_at_core_utest.c [hunk 18]
```diff
@@ -0,0 +1,932 @@
...
+
+    /* Check that CELLULAR_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATGetSpecificNextTok( & pString, pDelimiter, NULL );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+}
+
+/**
+ * @brief Test happy path for Cellular_ATGetSpecificNextTok to return CELLULAR_AT_SUCCESS and got expected results.
+ */
+void test_Cellular_ATGetSpecificNextTok_Happy_Path( void )
```
Comment (aggarg) [inline test/unit-test/cellular_at_core_utest.c:518]: This can be broken in two tests.

File: test/unit-test/cellular_at_core_utest.c [hunk 19]
```diff
@@ -0,0 +1,932 @@
...
+{
+    CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
+    const char * pDelimiter = ",";
+    char * pTokOutput;
+    char * pString = malloc( sizeof( char ) * ( strlen( CELLULAR_SAMPLE_STRING_TWO_DOUBLE_QUOTE ) + 1 ) );
```
Comment (aggarg) [inline test/unit-test/cellular_at_core_utest.c:523]: What is the use of this malloc? Is this memory leaked?

File: test/unit-test/cellular_at_core_utest.c [hunk 20]
```diff
@@ -0,0 +1,932 @@
...
+
+    /* Check that CELLULAR_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATHexStrToHex( pString, NULL, hexDataLen );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+}
+
+/**
+ * @brief Test happy path for Cellular_ATHexStrToHex to return CELLULAR_AT_SUCCESS and got expected results.
+ */
+void test_Cellular_ATHexStrToHex_Happy_Path( void )
```
Comment (aggarg) [inline test/unit-test/cellular_at_core_utest.c:580]: This can be broken in two too:
* test_Cellular_ATHexStrToHex_Happy_Path_Uppercase_Hex_String
* test_Cellular_ATHexStrToHex_Happy_Path_Lowecase_Hex_String

File: test/unit-test/cellular_at_core_utest.c [hunk 21]
```diff
@@ -0,0 +1,932 @@
...
+    uint8_t * const * const ppHexData;
+    uint16_t hexDataLen = 0;
+
+    /* Check that CELLULAR_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATHexStrToHex( pString, ppHexData, hexDataLen );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+
+    pString = CELLULAR_SAMPLE_HEX_STRING;
+    hexDataLen = 2;
+    /* Check that CELLULAR_BAD_PARAMETER is returned. */
```
Comment (aggarg) [inline test/unit-test/cellular_at_core_utest.c:613]: The comment says that check CELLULAR_BAD_PARAMETER is returned while assert checks CELLULAR_AT_NO_MEMORY. Also, this can be broken in 2 test cases. The idea is to just test one thing in one test case.

File: test/unit-test/cellular_at_core_utest.c [hunk 22]
```diff
@@ -0,0 +1,932 @@
...
+
+    /* Check that CELLULAR_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATIsStrDigit( pString, NULL );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+}
+
+/**
+ * @brief Test happy path for Cellular_ATIsStrDigit to return CELLULAR_AT_SUCCESS and got expected results.
+ */
+void test_Cellular_ATIsStrDigit_Happy_Path( void )
```
Comment (aggarg) [inline test/unit-test/cellular_at_core_utest.c:639]: test_Cellular_ATIsStrDigit_Simple_String
test_Cellular_ATIsStrDigit_String_ With_Null_Character

File: test/unit-test/cellular_at_core_utest.c [hunk 23]
```diff
@@ -0,0 +1,932 @@
...
+void test_Cellular_ATcheckErrorCode_Happy_Path( void )
+{
+    CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
+    char pString[] = CELLULAR_SAMPLE_SUCCESS_CODE_STRING;
+    char pStringError[] = CELLULAR_SAMPLE_ERROR_CODE_STRING;
+    char * ppKeyList[] = {"TEST1:SUCCESS", "TEST2:ERROR"};
+    size_t keyListLen = 2;
+    bool Result;
+
+    /* Check that CELLULAR_AT_BAD_PARAMETER is returned. */
```
Comment (aggarg) [inline test/unit-test/cellular_at_core_utest.c:702]: Comment and code are not in sync. In fact, these comments do not add any value as the code is self-explanatory. We can remove all these `Check that xxx is returned` comments.

File: test/unit-test/cellular_at_core_utest.c [hunk 24]
```diff
@@ -0,0 +1,932 @@
...
+    cellularStatus = Cellular_ATStrStartWith( pString, pPrefix, NULL );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+
+}
+
+/**
+ * @brief Test the error path for Cellular_ATStrStartWith to return CELLULAR_AT_BAD_PARAMETER.
+ */
+void test_Cellular_ATStrStartWith_Error_Path( void )
```
Comment (aggarg) [inline test/unit-test/cellular_at_core_utest.c:760]: test_Cellular_ATStrStartWith_Empty_String
test_Cellular_ATStrStartWith_Empty_Prefix

File: test/unit-test/cellular_at_core_utest.c [hunk 25]
```diff
@@ -0,0 +1,932 @@
...
+ */
+void test_Cellular_ATStrDup_Happy_Path( void )
+{
+    CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
+    char * pDst;
+    char pSrc[] = CELLULAR_SAMPLE_PREFIX_STRING_INPUT;
+
+    /* Check that CELLULAR_AT_SUCCESS is returned. */
+    cellularStatus = Cellular_ATStrDup( &pDst, pSrc );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_SUCCESS, cellularStatus );
```
Comment (aggarg) [inline test/unit-test/cellular_at_core_utest.c:862]: Verify that the string actually got duplicated by comparing the strings.

File: test/unit-test/cellular_at_core_utest.c [hunk 26]
```diff
@@ -0,0 +1,932 @@
...
+    cellularStatus = Cellular_ATIsPrefixPresent( pStr, &Result );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_SUCCESS, cellularStatus );
+    TEST_ASSERT_EQUAL( true, Result );
+    free( pStr );
+}
+
+/**
+ * @brief Test the error path for Cellular_ATIsPrefixPresent to return CELLULAR_AT_BAD_PARAMETER.
+ */
+void test_Cellular_ATIsPrefixPresent_Error_Path( void )
```
Comment (aggarg) [inline test/unit-test/cellular_at_core_utest.c:904]: Break in 3 testcases.

File: test/unit-test/cellular_config.h [hunk 27]
```diff
@@ -0,0 +1,57 @@
...
+ * http://aws.amazon.com/freertos
+ * http://www.FreeRTOS.org
+ */
+
+/**
+ * @file cellular_config.h
+ * @brief cellular config options.
+ */
+
+#ifndef __CELLULAR_CONFIG_H__
```
Comment (aggarg) [inline test/unit-test/cellular_config.h:31]:
```suggestion
#ifndef CELLULAR_CONFIG_H_

```

File: test/unit-test/cellular_platform.h [hunk 28]
```diff
@@ -0,0 +1,152 @@
...
+
+/*-----------------------------------------------------------*/
+
+
+/**
+ * @brief Cellular library platform event group APIs.
+ *
+ * Cellular library use platform event group for process synchronization.
+ *
+ * The EventGroup functions in the following link can be refernced as function prototype.
```
Comment (aggarg) [inline test/unit-test/cellular_platform.h:40]:
```suggestion
 * The EventGroup functions in the following link can be referenced as function prototype.

```

File: test/unit-test/cellular_platform.h [hunk 29]
```diff
@@ -0,0 +1,152 @@
...
+    void * pvDummy1[ 3 ];
+} StaticQueue_t;
+typedef StaticQueue_t StaticSemaphore_t;
+
+/**
+ * @brief Cellular library platform mutex APIs.
+ *
+ * Cellular library use platform mutex to protect resource.
+ *
+ * The IotMutex_ functions can be refernced as function prototype for
```
Comment (aggarg) [inline test/unit-test/cellular_platform.h:118]:
```suggestion
 * The IotMutex_ functions can be referenced as function prototype for

```

File: tools/cmock/project.yml [hunk 30]
```diff
@@ -0,0 +1,26 @@
+# Taken from amazon-freertos repository
```
Comment (aggarg) [inline tools/cmock/project.yml:1]: Please remove this comment.

File: test/unit-test/cellular_platform.h [hunk 31]
```diff
@@ -0,0 +1,152 @@
+/*
```
Comment (aggarg) [inline test/unit-test/cellular_platform.h:1]: Why is this file needed? Where is the implementation?

File: tools/cmock/create_test.cmake (hunk 1 above)
Comment (andysun2015) [reply tools/cmock/create_test.cmake:67]: Thanks.

File: test/unit-test/cellular_at_core_utest.c (hunk 4 above)
Comment (andysun2015) [reply test/unit-test/cellular_at_core_utest.c:206]: Could you explain this comment more specifically?

File: test/unit-test/cellular_at_core_utest.c (hunk 6 above)
Comment (andysun2015) [reply test/unit-test/cellular_at_core_utest.c]: Updated by [this commmit](https://github.com/FreeRTOS/Lab-Project-FreeRTOS-Cellular-Library/pull/8/commits/39dde55192c91c5dee1cd67cd43413ab47534e1f)

File: common/src/cellular_at_core.c (hunk 7 above)
Comment (andysun2015) [reply common/src/cellular_at_core.c:293]: Please take a look.
Update by [this commmit](https://github.com/FreeRTOS/Lab-Project-FreeRTOS-Cellular-Library/pull/8/commits/39dde55192c91c5dee1cd67cd43413ab47534e1f)

File: common/src/cellular_at_core.c (hunk 9 above)
Comment (andysun2015) [reply common/src/cellular_at_core.c:780]: No, I really don't know why it's modified, maybe it's caused by the editor.
Will roll back its modification.

File: test/unit-test/CMakeLists.txt (hunk 10 above)
Comment (andysun2015) [reply test/unit-test/CMakeLists.txt:2]: Sure.

File: test/unit-test/cellular_at_core_utest.c (hunk 12 above)
Comment (andysun2015) [reply test/unit-test/cellular_at_core_utest.c:72]: this sounds good, thanks.

File: test/unit-test/cellular_at_core_utest.c (hunk 19 above)
Comment (andysun2015) [reply test/unit-test/cellular_at_core_utest.c:525]: In some APIs, it will change the address of the pointer pString, that's why I back up its initial address to prevent free memory failed.
But in this case, this is not necessary, will fix it.

File: test/unit-test/cellular_at_core_utest.c (hunk 23 above)
Comment (andysun2015) [reply test/unit-test/cellular_at_core_utest.c:704]: Sure.

File: test/unit-test/cellular_at_core_utest.c (hunk 25 above)
Comment (andysun2015) [reply test/unit-test/cellular_at_core_utest.c:864]: Sure.

File: test/unit-test/cellular_config.h (hunk 27 above)
Comment (andysun2015) [reply test/unit-test/cellular_config.h:31]: Thanks for the suggestion.

File: test/unit-test/cellular_platform.h (hunk 28 above)
Comment (andysun2015) [reply test/unit-test/cellular_platform.h:40]: Thanks for the suggestion.

File: tools/cmock/project.yml (hunk 30 above)
Comment (andysun2015) [reply tools/cmock/project.yml:1]: Thanks for the suggestion.

File: test/unit-test/cellular_platform.h (hunk 31 above)
Comment (andysun2015) [reply test/unit-test/cellular_platform.h:1]: There will be some platform dependent code here, for cellular at command, the followings are needed:
#define Platform_Malloc    malloc
#define Platform_Free      free

And there will be other implementations updated with other test.

File: test/unit-test/cellular_at_core_utest.c (hunk 16 above)
Comment (andysun2015) [reply test/unit-test/cellular_at_core_utest.c:222]: sure.

File: test/unit-test/cellular_at_core_utest.c (hunk 17 above)
Comment (andysun2015) [reply test/unit-test/cellular_at_core_utest.c:273]: sounds good!

File: test/unit-test/cellular_at_core_utest.c (hunk 18 above)
Comment (andysun2015) [reply test/unit-test/cellular_at_core_utest.c:520]: sure, will divide this one into `test_Cellular_ATGetSpecificNextTok_WITH_DELIMTER_FIRST` and `test_Cellular_ATGetSpecificNextTok_WITHOUT_DELIMITER`.

File: test/unit-test/cellular_at_core_utest.c (hunk 22 above)
Comment (andysun2015) [reply test/unit-test/cellular_at_core_utest.c:641]: Thanks for the suggestion.

File: test/unit-test/cellular_at_core_utest.c (hunk 24 above)
Comment (andysun2015) [reply test/unit-test/cellular_at_core_utest.c:762]: Thanks for the suggestion.

**Review by aggarg — APPROVED** [review]

**Review by chinglee-iot — APPROVED** [review]
